src/engine/plugins/table.cpp

```diff
@@ -106,6 +132,30 @@ Status TablePlugin::HandleRequest(const RoutingAlgorithmsInterface &algorithms,
             {
                 const auto &table_index = row * num_destinations + column;
                 BOOST_ASSERT(table_index < result_tables_pair.first.size());
+                // Add decel/accel penalty to account for stoppage at the start/end of trip
+                if (stoppage_penalty_range >= 0 &&
+                    result_tables_pair.first[table_index] != MAXIMAL_EDGE_DURATION)
+                {
+                    const auto &source =
+                        snapped_phantoms[params.sources.empty() ? row : params.sources[row]];
+                    const auto &destination =
+                        snapped_phantoms[params.destinations.empty() ? column
+                                                                     : params.destinations[column]];
+
+                    // TODO: if phantom node is actual node, will the distance/duration be 0?
+
+                    auto source_penalty =
+                        stoppage_penalty(source.forward_distance / source.forward_duration,
+                                         params.min_stoppage_penalty,
+                                         stoppage_penalty_range);
```

Comment:
Oh bother - `forward` here is relative to the edge direction in the graph - and there's no guarantee that the `forward` direction was the one used in the actual route.  I can't think of an easy way to get that information at this location either....

Reply:
i wondered that :smile: 

does it mean that the duration/distance may be 0 depending on which one you pick?

Reply:
Yes.  In general I think I'm going to take back my "let's just adjust the table results after the fact" - there's not going to be an easy way to get it right here.  I think it will make more sense to adjust the phantom node durations before routing begins, then we don't have to worry about which one was picked.

Reply:
wont dont really care which direction they went, we only care the speed they went. can you have different durations depending on the direction you travel on the edge? i mean i know in real life its possible (but kind of absurd) but i mean in terms of the data model can we even have different speeds? also is the duration just the portion of the edge that was traveled or is it the full edge? 

seems like one of the directions had to be used, and if we can say an edge has the same speed in both directions we can just pick one to get the speed?